packages/commonwealth/server/util/getCommentDepth.ts

```diff
@@ -0,0 +1,29 @@
+import { DB } from 'server/models';
+
+// getCommentDepth recursively calculates the depth of a comment,
+// then returns if the depth exceeds max, and the depth level
+export const getCommentDepth = async (
+  models: DB,
+  comment,
+  maxIterations: number,
+  n?: number
+): Promise<[exceeded: boolean, depth: number]> => {
+  if (!n) {
+    n = 0;
+  }
+
+  if (!comment.parent_id) {
+    return [false, n];
+  }
+
+  if (n >= maxIterations) {
+    return [true, maxIterations];
+  }
+  const parentComment = await models.Comment.findOne({
```

Comment:
I wish there was a way to do this via postgres directly rather than recursively, but this is OK for now.

Reply:
Yeah that would be more performant. I'd need to write an integration test to feel safe about using a recursive SQL query.

FWIW, the updated code is at least an improvement upon the master branch because it applies a max number of iterations– previously, there was no limit.

Reply:
We'll address in a follow-up. No need for changes right now.